includes/filerepo/backend/GcsFileBackend.php

```diff
@@ -1,43 +1,82 @@
 <?php
 
+use Google\Cloud\Core\Exception\NotFoundException;
 use Google\Cloud\Storage\Bucket;
+use Google\Cloud\Storage\ObjectIterator;
 use Google\Cloud\Storage\StorageClient;
+use Google\Cloud\Storage\StorageObject;
 use Wikia\Logger\WikiaLogger;
 
 class GcsFileBackend extends FileBackendStore {
 
 	private $storage;
-
 	private $bucketName;
-
+	private $temporaryBucketName;
 	private $objectNamePrefix;
 
 	public function __construct( $config ) {
 		parent::__construct( $config );
-		$this->storage = new StorageClient( [
-			'keyFile' => $config['gcsCredentials']
-		] );
+		$this->storage = new StorageClient( [ 'keyFile' => $config['gcsCredentials'] ] );
 		$this->bucketName = $config['gcsBucket'];
+		$this->temporaryBucketName = $config['gcsTemporaryBucket'];
 		$this->objectNamePrefix = $config['gcsObjectNamePrefix'];
 	}
 
 	/**
-	 * Get a local copy on disk of the file at a storage path in the backend.
-	 * The temporary copy will have the same file extension as the source.
-	 * Temporary files may be purged when the file object falls out of scope.
-	 *
-	 * $params include:
-	 *     src    : source storage path
-	 *     latest : use the latest available data
-	 *
-	 * @param $params Array
-	 * @return TempFSFile|null Returns null on failure
+	 * Copied from SwiftFileBackend for compatibility. Might have no sense.
+	 * @see FileBackendStore::isValidContainerName()
+	 */
+	protected static function isValidContainerName( $container ) {
+		return preg_match( '/^[a-z0-9][a-z0-9-_.]{0,199}$/i', $container );
+	}
+
+	/**
+	 * Copied from SwiftFileBackend for compatibility. Might have no sense.
+	 * @see FileBackendStore::resolveContainerPath()
+	 * @param $container
+	 * @param $relStoragePath
+	 * @return string|null
+	 */
+	protected function resolveContainerPath( $container, $relStoragePath ) {
+		if ( !mb_check_encoding( $relStoragePath, 'UTF-8' ) ) { // mb_string required by CF
+			return null; // not UTF-8, makes it hard to use CF and the swift HTTP API
+		} elseif ( strlen( urlencode( $relStoragePath ) ) > 1024 ) {
```

Comment:
👍

Reply:
as it's utf-8, should we use `mb_strlen` (character count)?